setup/res/installer.js

```diff
@@ -0,0 +1,31 @@
+(function($){
+  /**
+   * Enable or disable an error message.
+   *
+   * <p id="my-error-message">The world is one fire.</p>
+   * Ex: $('#my-error-message').toggleError(false)
+   *
+   * @param bool isError
+   */
+  $.fn.toggleError = function (isError) {
+    this.toggleClass('install-validate-ok', !isError)
+      .toggleClass('install-validate-bad', isError)
+      .toggleClass('error', isError);
+
+    var errors = $('.install-validate-bad');
+    $('#install_button').prop('disabled', errors.length > 0);
+    return this;
+  };
+
+  /**
+   * Ex: $('.watch-these').useValidator(function(){
+   *   $('#some-error-message').toggleError(booleanExpression);
+   * })
+   * @param cb
+   */
+  $.fn.useValidator = function(cb) {
+    cb();
+    this.on('change', cb);
+    return this;
+  };
+})($);
```

Comment:
is this right?

Reply:
@seamuslee001 You're asking if the closure is properly formed? Well, `$` seemed to bind correctly.

Note that the setup page doesn't have all the normal services of a Civi page, so we work from first principles. JS on the setup page loads like this:

```html
<script type="text/javascript" src="http://dmaster.bknix:8001/sites/all/modules/civicrm/bower_components/jquery/dist/jquery.min.js"></script>
<script type="text/javascript" src="http://dmaster.bknix:8001/sites/all/modules/civicrm/setup/res/installer.js"></script>
<script type="text/javascript">
  window.csj$ = jQuery.noConflict();
</script>
```

Which means the JS file should be written in the style of a vanilla jQuery plugin.

Double-checking some other examples (`jquery.jeditable.js`, `jquery.notify.js`, `jquery.tokeninput.js`, etc), it seems more typical for vanilla-style jQuery plugins to take the input as `jQuery`. So on style grounds this snippet is not canonical.

I've pushed an update to make it clearer that this file works like a vanilla jQuery plugin.